parallelize check, cleanup config, better fot logging

grafnu opened the pull request.

didovesei commented on daq/runner.py:
```diff
@@ -926,18 +926,26 @@ def _send_device_result(self, mac, test_results):
+                failed = True
+
+            module_tests = module_result.get('tests', {})
+            for test_name in module_tests:
```
Maybe `for test_name, test_result  in module_tests`.items() is better?

didovesei commented on daq/runner.py:
```diff
@@ -926,18 +926,26 @@ def _send_device_result(self, mac, test_results):
         self._device_result_client.send_device_result(mac, device_result)
 
     def _calculate_device_result(self, test_results):
+        failed = False
```
I am not sure why we always check all the three items: exception, return code, and module test results, since one failure is already enough to indicate that the test has failed. Is it because we want to have a more verbose logging on the test results?

grafnu commented:
Yes -- I wanted more comprehensive logging; that was the main reason for
the change.  There's no real functional difference.




On Fri, Feb 12, 2021 at 2:44 AM Yufeng Duan <notifications@github.com>
wrote:

> *@didovesei* commented on this pull request.
> ------------------------------
>
> In daq/runner.py
> <https://github.com/faucetsdn/daq/pull/766#discussion_r575134471>:
>
> > @@ -926,18 +926,26 @@ def _send_device_result(self, mac, test_results):
>          self._device_result_client.send_device_result(mac, device_result)
>
>      def _calculate_device_result(self, test_results):
> +        failed = False
>
> I am not sure why we always check all the three items: exception, return
> code, and module test results, since one failure is already enough to
> indicate that the test has failed. Is it because we want to have a more
> verbose logging on the test results?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/faucetsdn/daq/pull/766#pullrequestreview-589349066>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIEPD6HUMJJGB22A52HZ5TS6UBCFANCNFSM4XQEEPIA>
> .
>

henry54809 commented on cmd/build:
```diff
@@ -25,7 +25,6 @@ DOCKER_IMAGE_LIST=docker_images.txt
 DOCKER_IMAGE_VER=docker_images.ver
 
 cd $ROOT
-source etc/FILES_MAPPING
```
Why are these lines removed?

henry54809 commented on etc/config_base.sh:
```diff
@@ -1,5 +1,7 @@
 # Script file included by all setup scripts to load local config.
 
+source etc/FILES_MAPPING
```
Putting this here won’t work with the Debian package files locations since this script is also run and the env vars will not be the right ones.

grafnu commented:
The goal was to consolidate it into the shared configuration mechanism --
since it's a form of configuration. Trying to maintain the abstraction and
not bleed "etc/FILES_MAPPING" everywhere.  Seems like debian might be a
special case, but we should sort through that separately.

On Fri, Feb 12, 2021 at 8:23 AM henry54809 <notifications@github.com> wrote:

> *@henry54809* commented on this pull request.
> ------------------------------
>
> In cmd/build
> <https://github.com/faucetsdn/daq/pull/766#discussion_r575342237>:
>
> > @@ -25,7 +25,6 @@ DOCKER_IMAGE_LIST=docker_images.txt
>  DOCKER_IMAGE_VER=docker_images.ver
>
>  cd $ROOT
> -source etc/FILES_MAPPING
>
> Why are these lines removed?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/faucetsdn/daq/pull/766#pullrequestreview-589618949>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIEPD7QAJYPAPZHEPP7UBDS6VIXDANCNFSM4XQEEPIA>
> .
>

grafnu commented:
OK -- I"ll have a look more closely at the debian packages. I saw the
explicit reference to debian/FILES_MAPPING and probably just made a bad
assumption about how it worked. I think the right solution (if I can figure
it out) is to make sure that the FILES_MAPPING that's installed with debian
is the right one (or doesn't exist) so that it doesn't cause problems.

On Fri, Feb 12, 2021 at 8:25 AM henry54809 <notifications@github.com> wrote:

> *@henry54809* commented on this pull request.
> ------------------------------
>
> In etc/config_base.sh
> <https://github.com/faucetsdn/daq/pull/766#discussion_r575343899>:
>
> > @@ -1,5 +1,7 @@
>
>  # Script file included by all setup scripts to load local config.
>
>
>
> +source etc/FILES_MAPPING
>
>
> Putting this here won’t work with the Debian package files locations since
> this script is also run and the env vars will not be the right ones.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/faucetsdn/daq/pull/766#pullrequestreview-589620951>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIEPDZF323BGC6C7WOLAMTS6VI7JANCNFSM4XQEEPIA>
> .
>

henry54809 commented:
How about in setup.py, replace the line "source etc/FILES_MAPPING" in etc/config_base.sh with the content of debian/FILES_MAPPING?
The problem I was encountering was it is not known if DAQ is in a package form in scripts without the env vars. 

> The goal was to consolidate it into the shared configuration mechanism -- since it's a form of configuration. Trying to maintain the abstraction and not bleed "etc/FILES_MAPPING" everywhere. Seems like debian might be a special case, but we should sort through that separately.
> […](#)
> On Fri, Feb 12, 2021 at 8:23 AM henry54809 ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In cmd/build <[#766 (comment)](https://github.com/faucetsdn/daq/pull/766#discussion_r575342237)>: > @@ -25,7 +25,6 @@ DOCKER_IMAGE_LIST=docker_images.txt DOCKER_IMAGE_VER=docker_images.ver cd $ROOT -source etc/FILES_MAPPING Why are these lines removed? — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub <[#766 (review)](https://github.com/faucetsdn/daq/pull/766#pullrequestreview-589618949)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAIEPD7QAJYPAPZHEPP7UBDS6VIXDANCNFSM4XQEEPIA> .

grafnu commented:
Ok, I updated and tested a solution that works with debian. Basically it only includes the file mappings if they're not already defined (which is the case with debian). I also moved the common config thing to bin/ (from etc/) -- a bit cleaner since then it's not special cased.

grafnu commented:
PTAL (mainly Haoli for the debian part)
